packages/dashboard/src/customization/hooks/useRefreshRate.ts

```diff
@@ -0,0 +1,13 @@
+import { useSelector } from 'react-redux';
+import { DashboardState } from '~/store/state';
+
+export const useRefreshRate = () => {
+  const refreshRate = useSelector(
+    (state: DashboardState) =>
+      state.dashboardConfiguration.querySettings.refreshRate
+  );
+
+  return {
+    refreshRate,
+  };
+};
```

Comment:
Improve this by also encapsulating the dispatch to update state. You could return a function called `updateRate` or similar.